bb/src/tools/release.clj

```diff
@@ -0,0 +1,43 @@
+(ns tools.release
+  (:require [babashka.fs :as fs]
+            [borkdude.gh-release-artifact.internal :as gh]
+            [tools.build :refer [jar-path]]
+            [tools.version :as version])
+  (:import (clojure.lang ExceptionInfo)))
+
+(defn fib [a b]
+  (lazy-seq (cons a (fib b (+ a b)))))
+
+(defn retry-with-fib-backoff [retries exec-fn test-fn]
+  (loop [idle-times (take retries (fib 1 2))]
+    (let [result (exec-fn)]
+      (if (test-fn result)
+        (when-let [sleep-ms (first idle-times)]
+          (println "Returned: " result)
+          (println "Retrying with remaining back-off times (in s): " idle-times)
+          (Thread/sleep (* 1000 sleep-ms))
+          (recur (rest idle-times)))
+        result))))
+
+(defn try-release [config]
+  (try (gh/overwrite-asset {:org (:org config)
+                            :repo (name (:lib config))
+                            :tag (version/string config)
+                            :commit (gh/current-commit)
+                            :file (jar-path config)
+                            :content-type "application/java-archive"
+                            :draft false})
```

Comment:
sure you want to use the `internal` ns? guess it's not for public usage so it might change in future. you could copy the `current-commit`-fn into the init-part in the bb.edn and we'll have the patch-release-version also available in the whole build process. might come in handy later.

Reply:
I surely never wanted to use an internal function, but it seemed to me like the function moved there for unknown reasons. I must have been pretty blind though when I checked this, because the function still exists in the old namespace and there is another one there that seems to be the one that is to be used. Thanks for catching this...

Reply:
I have changed this now